src/test/java/net/smartcosmos/dao/objects/impl/ObjectPersistenceServiceTest.java

```diff
@@ -108,8 +113,208 @@ public void getObjects() throws Exception {
+        response = objectPersistenceService
+            .findByQueryParameters(accountUrn, queryParams);
+        assertTrue(response.size() == 9);
+
+        queryParams.put(ObjectPersistenceService.OBJECT_URN_LIKE, "objectUrnQueryParams1");
+        response = objectPersistenceService
+            .findByQueryParameters(accountUrn, queryParams);
+        assertTrue(response.size() == 3);
+
+        queryParams.put(ObjectPersistenceService.OBJECT_URN_LIKE, "objectUrnQueryParams11");
+        response = objectPersistenceService
+            .findByQueryParameters(accountUrn, queryParams);
+        assertTrue(response.size() == 1);
+
+        queryParams.put(ObjectPersistenceService.OBJECT_URN_LIKE, "objectUrnQueryParams99");
+        response = objectPersistenceService
+            .findByQueryParameters(accountUrn, queryParams);
+        assertTrue(response.size() == 0);
+
+        queryParams.put(ObjectPersistenceService.OBJECT_URN_LIKE, "bjectUrnQueryParams");
+        response = objectPersistenceService
+            .findByQueryParameters(accountUrn, queryParams);
+        assertTrue(response.size() == 0);
+
+        queryParams.put(ObjectPersistenceService.OBJECT_URN_LIKE, "objectUrnQueryParams");
+        queryParams.put(ObjectPersistenceService.TYPE, "type");
+        response = objectPersistenceService
+            .findByQueryParameters(accountUrn, queryParams);
+        assertTrue(response.size() == 6);
+
+        queryParams.put(ObjectPersistenceService.TYPE, "type o");
+        response = objectPersistenceService
+            .findByQueryParameters(accountUrn, queryParams);
+        assertTrue(response.size() == 3);
+
+        queryParams.put(ObjectPersistenceService.TYPE, "type t");
+        response = objectPersistenceService
+            .findByQueryParameters(accountUrn, queryParams);
+        assertTrue(response.size() == 3);
+
+        queryParams.put(ObjectPersistenceService.TYPE, "type z");
+        response = objectPersistenceService
+            .findByQueryParameters(accountUrn, queryParams);
+        assertTrue(response.size() == 0);
+
+        queryParams.put(ObjectPersistenceService.TYPE, "ype");
+        response = objectPersistenceService
+            .findByQueryParameters(accountUrn, queryParams);
+        assertTrue(response.size() == 0);
+
+        queryParams.remove(ObjectPersistenceService.TYPE);
+        queryParams.put(ObjectPersistenceService.MONIKER_LIKE, "moniker");
+        response = objectPersistenceService
+            .findByQueryParameters(accountUrn, queryParams);
+        assertTrue(response.size() == 3);
+
+        queryParams.put(ObjectPersistenceService.MONIKER_LIKE, "moniker o");
+        response = objectPersistenceService
+            .findByQueryParameters(accountUrn, queryParams);
+        assertTrue(response.size() == 1);
+
+        queryParams.put(ObjectPersistenceService.MONIKER_LIKE, "moniker t");
```

Comment:
That's all I'm testing per string, plus verifying that it's doing an actual startsWith, thus "oniker". OK, there are a couple of places where I'm testing two different subsets. I can remove those if you want.